src/main/java/com/ksm/wstbackoffice/controller/CountryController.java

```diff
@@ -44,4 +47,16 @@ public CountryController(CountryService countryService) {
         return ResponseEntity.status(HttpStatus.OK)
                 .body(country);
     }
+
+    @GetMapping("/filter-name")
+    public ResponseEntity<Map<String, Map<String, ?>>> findAllBy(@RequestParam("nameStartsWithFilters") Collection<String> nameStartsWithFilters) {
+        if (nameStartsWithFilters.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+        }
+
+        Map<String, Map<String, ?>> filteredCountries = countryService.findAllBy(nameStartsWithFilters);
```

Comment:
Why do you need filteredCountries variable?

Reply:
done